transaction/core/src/ring_ct/rct_bulletproofs.rs

```diff
@@ -764,78 +813,6 @@ impl SignatureRctBulletproofs {
     }
 }
 
-/// Sign, with optional check for inputs = outputs.
-///
-/// # Arguments
-/// * `block_version` - This may influence details of the signature
-/// * `message` - The messages to be signed, e.g. Hash(TxPrefix).
-/// * `rings` - One or more rings of one-time addresses and amount commitments,
-///   with secrets for the real input
-/// * `output_secrets` - Output secret for each output amount commitment.
-/// * `fee` - Amount of the implicit fee output.
-/// * `check_value_is_preserved` - If true, check that the value of inputs
-///   equals value of outputs.
-/// * `rng` - randomness
-fn sign_with_balance_check<CSPRNG: RngCore + CryptoRng, S: RingSigner + ?Sized>(
```

Comment:
I removed this because at this point, it is more logical for it to be called `SigningData.sign`.
I think that may be the pathway that I envision hardware wallets actually using.